src/main/java/htsjdk/samtools/SAMTextHeaderCodec.java

```diff
@@ -136,7 +136,6 @@ private String advanceLine() {
             return null;
         }
         mCurrentLine = mReader.readLine();
-        textHeader.append(mCurrentLine).append('\n');
```

Comment:
It's set in SAMFileHeader.java if there's an error parsing the header.  👎

Reply:
Point me to the unit test that covers that situation.

Reply:
That kind of attitude is not going to earn any internet points here.  Not everything is covered by a unit test (fortunately or unfortunately). Taking 10 seconds to search the source your changing would have revealed it.  I'll let @tfenne @jacarey and @lbergelson to take over from here.

Reply:
> That kind of attitude is not going to earn any internet points here. 

Based on past experience with you and others in this repository I felt the only way to fight snark was with more snark.  I apologize.

Let me try again, how about instead of collecting the entire header in RAM on the off chance that it might need to be reported in an error condition, track only the line number, and then include that with the offending header line when reporting the error?

Reply:
@heuermh I have a bolder proposal:
1. Deprecate `getTextHeader()` in `SAMFileHeader` and change it to `public String getTextHeader() { return null; }`.  I see only Picard's `ViewSam` tool use this method, and it does the right thing if `null` is returned.
2. Remove any reference to `textHeader` in `SAMTextHeaderCodec`.  We don't need it to report errors, since we store line numbers and the error message using `reportErrorParsingLine`.

Reply:
@heuermh Thank you for identifying and reporting the problem.  It definitely looks like this is a vestigial method we don't need.  I checked gatk as well and didn't find any uses.  

I like @nh13's proposal to fix the problem.  Much better to deprecate and remove the unnecessary stuff than just to disconnect it and leave it hanging around in the code.  @heuermh would you be able to revise the PR?

Reply:
It appears that we call `SAMFileHeader.getTextHeader` in Hadoop-BAM at [SAMRecordReader.java#L196](https://github.com/HadoopGenomics/Hadoop-BAM/blob/f2e9595d35ea2118e3c56ebeedc29cd432ed4b9e/src/main/java/org/seqdoop/hadoop_bam/SAMRecordReader.java#L196), in a class full of workarounds for various issues with SAMFileReader.  I'll check if these changes cause trouble there.

What is htsjdk's policy on deprecation, versioning, and binary compatibility?  Typically I would annotate as @deprecated in a minor or point release without code change and then remove on next major release.

I'll update this PR as necessary, squash and rebase with force pushes ok?

Reply:
Maybe rather than returning `null` the deprecated method could `return getSAMString()` instead, which will reconstruct the header as a String from the internal state?  Otherwise I agree with @nh13's approach.

@heuermh: we tend to prefer not squashing until the final merge if you don't mind please.

Reply:
@heuermh We don't have a really well defined deprecation policy.  In general we try to deprecate without breaking the method and let that hang out for a few minor versions before removing any functionality.  We avoid any incompatible changes between patch releases, but minor releases change the api in non-backward's compatible ways.  

Since this method already has a lot of caveats about when it returns a value and when it returns null, callers should already be able to handle the null safely without loss of functionality.  In any case though, @tfenne's suggestion is better than returning null.

Reply:
@heuermh are you going to get back to this? it seems like the way forward was resolved a few months back.

Thanks!